### ISSUE TITLE
Customizing Researchkit View look

### ISSUE THREAD

**Issue by nicholas12345** [ResearchKit/ResearchKit#508]
Hi, is there a way to customize how for example survey form looks? To be precise, it would be neat to be able to change background, text colors, background of cells etc. I failed to find any traces of that in documentation, so any help would be super helpful: tips, demo code, snippet, hint...

Is there maybe two themes of Researchkit (like navbar in iOS provides black and white)? 

Thanks!

**Comment by YuanZhu-apple** [comment]
RK support `tintColor` customization, for example:
```
[UIView appearance].tintColor = [UIColor redColor]; 
```

**Comment by nicholas12345** [comment]
Ok that works, however I do not want to affect any other views, just where ResearchKit resides. For example, Master detail: from master I am directed to detail where Researchkit Survey form resides. Changing UIView.appearance() in
`func taskViewController(ORKTaskViewController, stepViewControllerWillAppear: ORKStepViewController)`
will result in changes in Researchkit but in any other view as well once I return to master.

How to access properties of RK elements? For eaxmple to set tablecell background to clearColor, for example?

Many thanks!

**Comment by YuanZhu-apple** [comment]
To only apply `tintColor` to RK:
`[UIView appearanceWhenContainedIn:[ORKTaskViewController class], nil].tintColor = [UIColor redColor];` 

Similarly, you can use this customize other properties, but use it with caution, since we only tested `tintColor`.

**Comment by nicholas12345** [comment]
Ok, great. But unfortunately in iOS 8 those methods, that I believe were marked deprecated in iOS 7, are removed from Swift libs. :-(

**Comment by YuanZhu-apple** [comment]
Yeah, use  `appearanceWhenContainedInInstancesOfClasses:` instead.

**Comment by nicholas12345** [comment]
Again correct, but for iOS 9. :-D Unfortunately, it seems that Swift iOS 8 cannot achieve this, unless you know some other magic?

**Comment by YuanZhu-apple** [comment]
```
+ (instancetype)appearanceWhenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(5_0, 9_0, "Use +appearanceWhenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceWhenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes NS_AVAILABLE_IOS(9_0);

+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait NS_AVAILABLE_IOS(8_0);
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(8_0, 9_0, "Use +appearanceForTraitCollection:whenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes  NS_AVAILABLE_IOS(9_0);
+ (instancetype)appearanceWhenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(5_0, 9_0, "Use +appearanceWhenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceWhenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes NS_AVAILABLE_IOS(9_0);

+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait NS_AVAILABLE_IOS(8_0);
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(8_0, 9_0, "Use +appearanceForTraitCollection:whenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes  NS_AVAILABLE_IOS(9_0);+ (instancetype)appearanceWhenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(5_0, 9_0, "Use +appearanceWhenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceWhenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes NS_AVAILABLE_IOS(9_0);

+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait NS_AVAILABLE_IOS(8_0);
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedIn:(nullable Class <UIAppearanceContainer>)ContainerClass, ... NS_REQUIRES_NIL_TERMINATION NS_DEPRECATED_IOS(8_0, 9_0, "Use +appearanceForTraitCollection:whenContainedInInstancesOfClasses: instead");
+ (instancetype)appearanceForTraitCollection:(UITraitCollection *)trait whenContainedInInstancesOfClasses:(NSArray<Class <UIAppearanceContainer>> *)containerTypes  NS_AVAILABLE_IOS(9_0);
```

Can you double check? Those are deprecated in iOS 9.

**Comment by nicholas12345** [comment]
Well, in 
`func taskViewController(ORKTaskViewController, stepViewControllerWillAppear: ORKStepViewController)`
I tried to call the way you suggested, but autocomplete gives only what is in attach. I tried on any ViewController to call it, but still same. In the end, I also tried to type it manually, but it complains that  
`'UIView.Type' does not have a member named 'appearanceWhenContainednInstancesOfClasses'`

What really bothers me is that app is mostly with dark background so navigation items are white color, but when RK Survey opens up, its "Cancel/Done" button on top right is white, and on white background it is invisible. So idea is to either change the background or a button (ideally for now might be button). But unfortunately, even after toying few days with it, I couldn't find any solution, thus asking here. So, I googled a bit and found this answer: http://stackoverflow.com/a/27807417/3481859
Apparently, Swift does not have those method that are in Objective C (they are added now in latest one?). So, I believe I am sure I cannot solve it the way you suggested, unfortunately. :-(

If you have any more hints, great. Thanks a lot for all the help so far, you rock.
